pinDocker.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+func pinDocker(action, jobName, inputYaml string) string {
```

Comment:
@Devils-Knight please create a test file and write test cases. The test cases should cover ghcr.io, docker hub, and may be other providers

Reply:
@varunsh-coder I am unable to find what are the `requests` made by `img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))` this function call.

[Here](https://github.com/google/go-containerregistry/tree/main/pkg/v1/remote) are the details of this function call.